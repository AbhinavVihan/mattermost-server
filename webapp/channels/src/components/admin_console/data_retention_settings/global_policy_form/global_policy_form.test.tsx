// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import GlobalPolicyForm from 'components/admin_console/data_retention_settings/global_policy_form/global_policy_form';
import {renderWithIntl} from 'tests/react_testing_utils';
import {Provider} from 'react-redux';
import store from 'stores/redux_store';
import {BrowserRouter} from 'react-router-dom';

describe('components/PluginManagement', () => {
    const defaultProps = {
        config: {
            DataRetentionSettings: {
                EnableMessageDeletion: true,
                EnableFileDeletion: true,
                MessageRetentionDays: 60,
                FileRetentionDays: 40,
                DeletionJobStartTime: '10:00',
            },
        },
        actions: {
            updateConfig: jest.fn(),
            setNavigationBlocked: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const props = {...defaultProps};
        const wrapper = renderWithIntl(<BrowserRouter><Provider store={store}><GlobalPolicyForm {...props}/></Provider></BrowserRouter>);
        expect(wrapper).toMatchSnapshot();
    });
});
