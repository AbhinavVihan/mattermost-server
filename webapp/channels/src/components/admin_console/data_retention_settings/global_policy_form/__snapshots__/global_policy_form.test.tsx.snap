// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PluginManagement should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper--fixed DataRetentionSettings"
      >
        <div
          class="admin-console__header with-back"
        >
          <div>
            <a
              class="fa fa-angle-left back"
              href="/admin_console/compliance/data_retention_settings"
            />
            Global Retention Policy
          </div>
        </div>
        <div
          class="admin-console__wrapper"
        >
          <div
            class="admin-console__content"
          >
            <div
              class="Card console expanded"
            >
              <div
                class="Card__body expanded expanding"
                style="height: 0px;"
              >
                <div
                  class="global_policy"
                >
                  <p>
                    Applies to all teams and channels, but does not apply to custom retention policies.
                  </p>
                  <div
                    id="global_direct_message_dropdown"
                  >
                    <div
                      class="DropdownInput hybrid_container"
                      style="width: 100%;"
                    >
                      <fieldset
                        class="Input_fieldset Input_fieldset___legend Input_fieldset___split"
                      >
                        <legend
                          class="Input_legend Input_legend___focus"
                        >
                          Channel & direct message retention
                        </legend>
                        <div
                          class="Input_wrapper input_hybrid_wrapper showInput"
                          style="max-width: 10000px;"
                        >
                          <input
                            class="Input form-control"
                            id="channel_message_retention_input"
                            name="Input_channel_message_retention"
                            placeholder="Channel & direct message retention"
                            type="number"
                            value="60"
                          />
                        </div>
                        <div
                          class="Input_wrapper dropdown_hybrid_wrapper"
                          style="width: 90px;"
                        >
                          <div
                            class="Input Input__focus css-2b097c-container"
                            id="DropdownInput_channel_message_retention"
                          >
                            <div
                              class="DropdownInput__controlContainer"
                            >
                              <div
                                class="channel_message_retention_dropdown__control css-1nr98fh-control"
                              >
                                <div
                                  class="channel_message_retention_dropdown__value-container channel_message_retention_dropdown__value-container--has-value css-1hwfws3"
                                >
                                  <div
                                    class="channel_message_retention_dropdown__single-value css-1uccc91-singleValue"
                                  >
                                    <span
                                      class="option_days"
                                    >
                                      Days
                                    </span>
                                  </div>
                                  <input
                                    class="css-62g3xt-dummyInput"
                                    id="react-select-2-input"
                                    readonly=""
                                    tabindex="0"
                                    value=""
                                  />
                                </div>
                                <div
                                  class="DropdownInput__indicatorsContainer"
                                >
                                  <div
                                    class="channel_message_retention_dropdown__indicators css-1hb7zxy-IndicatorsContainer"
                                  >
                                    <i
                                      class="icon icon-chevron-down"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    id="global_file_dropdown"
                  >
                    <div
                      class="DropdownInput hybrid_container"
                      style="width: 100%;"
                    >
                      <fieldset
                        class="Input_fieldset Input_fieldset___legend Input_fieldset___split"
                      >
                        <legend
                          class="Input_legend Input_legend___focus"
                        >
                          File retention
                        </legend>
                        <div
                          class="Input_wrapper input_hybrid_wrapper showInput"
                          style="max-width: 10000px;"
                        >
                          <input
                            class="Input form-control"
                            id="file_retention_input"
                            name="Input_file_retention"
                            placeholder="File retention"
                            type="number"
                            value="40"
                          />
                        </div>
                        <div
                          class="Input_wrapper dropdown_hybrid_wrapper"
                          style="width: 90px;"
                        >
                          <div
                            class="Input Input__focus css-2b097c-container"
                            id="DropdownInput_file_retention"
                          >
                            <div
                              class="DropdownInput__controlContainer"
                            >
                              <div
                                class="file_retention_dropdown__control css-1nr98fh-control"
                              >
                                <div
                                  class="file_retention_dropdown__value-container file_retention_dropdown__value-container--has-value css-1hwfws3"
                                >
                                  <div
                                    class="file_retention_dropdown__single-value css-1uccc91-singleValue"
                                  >
                                    <span
                                      class="option_days"
                                    >
                                      Days
                                    </span>
                                  </div>
                                  <input
                                    class="css-62g3xt-dummyInput"
                                    id="react-select-3-input"
                                    readonly=""
                                    tabindex="0"
                                    value=""
                                  />
                                </div>
                                <div
                                  class="DropdownInput__indicatorsContainer"
                                >
                                  <div
                                    class="file_retention_dropdown__indicators css-1hb7zxy-IndicatorsContainer"
                                  >
                                    <i
                                      class="icon icon-chevron-down"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="admin-console-save"
        >
          <button
            class="save-button btn"
            data-testid="saveSetting"
            disabled=""
            id="saveSetting"
            type="submit"
          >
            <span>
              Save
            </span>
          </button>
          <a
            class="cancel-button"
            href="/admin_console/compliance/data_retention_settings"
          >
            Cancel
          </a>
          
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper--fixed DataRetentionSettings"
    >
      <div
        class="admin-console__header with-back"
      >
        <div>
          <a
            class="fa fa-angle-left back"
            href="/admin_console/compliance/data_retention_settings"
          />
          Global Retention Policy
        </div>
      </div>
      <div
        class="admin-console__wrapper"
      >
        <div
          class="admin-console__content"
        >
          <div
            class="Card console expanded"
          >
            <div
              class="Card__body expanded expanding"
              style="height: 0px;"
            >
              <div
                class="global_policy"
              >
                <p>
                  Applies to all teams and channels, but does not apply to custom retention policies.
                </p>
                <div
                  id="global_direct_message_dropdown"
                >
                  <div
                    class="DropdownInput hybrid_container"
                    style="width: 100%;"
                  >
                    <fieldset
                      class="Input_fieldset Input_fieldset___legend Input_fieldset___split"
                    >
                      <legend
                        class="Input_legend Input_legend___focus"
                      >
                        Channel & direct message retention
                      </legend>
                      <div
                        class="Input_wrapper input_hybrid_wrapper showInput"
                        style="max-width: 10000px;"
                      >
                        <input
                          class="Input form-control"
                          id="channel_message_retention_input"
                          name="Input_channel_message_retention"
                          placeholder="Channel & direct message retention"
                          type="number"
                          value="60"
                        />
                      </div>
                      <div
                        class="Input_wrapper dropdown_hybrid_wrapper"
                        style="width: 90px;"
                      >
                        <div
                          class="Input Input__focus css-2b097c-container"
                          id="DropdownInput_channel_message_retention"
                        >
                          <div
                            class="DropdownInput__controlContainer"
                          >
                            <div
                              class="channel_message_retention_dropdown__control css-1nr98fh-control"
                            >
                              <div
                                class="channel_message_retention_dropdown__value-container channel_message_retention_dropdown__value-container--has-value css-1hwfws3"
                              >
                                <div
                                  class="channel_message_retention_dropdown__single-value css-1uccc91-singleValue"
                                >
                                  <span
                                    class="option_days"
                                  >
                                    Days
                                  </span>
                                </div>
                                <input
                                  class="css-62g3xt-dummyInput"
                                  id="react-select-2-input"
                                  readonly=""
                                  tabindex="0"
                                  value=""
                                />
                              </div>
                              <div
                                class="DropdownInput__indicatorsContainer"
                              >
                                <div
                                  class="channel_message_retention_dropdown__indicators css-1hb7zxy-IndicatorsContainer"
                                >
                                  <i
                                    class="icon icon-chevron-down"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </fieldset>
                  </div>
                </div>
                <div
                  id="global_file_dropdown"
                >
                  <div
                    class="DropdownInput hybrid_container"
                    style="width: 100%;"
                  >
                    <fieldset
                      class="Input_fieldset Input_fieldset___legend Input_fieldset___split"
                    >
                      <legend
                        class="Input_legend Input_legend___focus"
                      >
                        File retention
                      </legend>
                      <div
                        class="Input_wrapper input_hybrid_wrapper showInput"
                        style="max-width: 10000px;"
                      >
                        <input
                          class="Input form-control"
                          id="file_retention_input"
                          name="Input_file_retention"
                          placeholder="File retention"
                          type="number"
                          value="40"
                        />
                      </div>
                      <div
                        class="Input_wrapper dropdown_hybrid_wrapper"
                        style="width: 90px;"
                      >
                        <div
                          class="Input Input__focus css-2b097c-container"
                          id="DropdownInput_file_retention"
                        >
                          <div
                            class="DropdownInput__controlContainer"
                          >
                            <div
                              class="file_retention_dropdown__control css-1nr98fh-control"
                            >
                              <div
                                class="file_retention_dropdown__value-container file_retention_dropdown__value-container--has-value css-1hwfws3"
                              >
                                <div
                                  class="file_retention_dropdown__single-value css-1uccc91-singleValue"
                                >
                                  <span
                                    class="option_days"
                                  >
                                    Days
                                  </span>
                                </div>
                                <input
                                  class="css-62g3xt-dummyInput"
                                  id="react-select-3-input"
                                  readonly=""
                                  tabindex="0"
                                  value=""
                                />
                              </div>
                              <div
                                class="DropdownInput__indicatorsContainer"
                              >
                                <div
                                  class="file_retention_dropdown__indicators css-1hb7zxy-IndicatorsContainer"
                                >
                                  <i
                                    class="icon icon-chevron-down"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="admin-console-save"
      >
        <button
          class="save-button btn"
          data-testid="saveSetting"
          disabled=""
          id="saveSetting"
          type="submit"
        >
          <span>
            Save
          </span>
        </button>
        <a
          class="cancel-button"
          href="/admin_console/compliance/data_retention_settings"
        >
          Cancel
        </a>
        
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
